query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -323,34 +328,31 @@ private String convertExpressionToMapValueColumn(Expression expression) {
     throw new IllegalArgumentException("operator supports multi value column only");
   }
 
-  private LiteralConstant[] convertExpressionToMapLiterals(Expression expression) {
-    LiteralConstant[] literals = new LiteralConstant[2];
+  private List<LiteralConstant> convertExpressionToMapLiterals(Expression expression) {
+    List<String> literals = new ArrayList<>(List.of("", ""));
     if (expression.getValueCase() == LITERAL) {
       LiteralConstant value = expression.getLiteral();
-      if (value.getValue().getValueType() == ValueType.STRING_ARRAY) {
-        for (int i = 0; i < 2 && i < value.getValue().getStringArrayCount(); i++) {
-          literals[i] =
-              LiteralConstant.newBuilder()
-                  .setValue(
-                      Value.newBuilder().setString(value.getValue().getStringArray(i)).build())
-                  .build();
-        }
+      if (value.getValue().getValueType() == ValueType.STRING_ARRAY
+          && value.getValue().getStringArrayCount() == 2) {
+        literals.set(0, value.getValue().getStringArray(0));
```

Comment:
nit: if we're going to keep it unstructured and use a list, might as well use the same index constants to write this as we use to read it.

Reply:
Didn't get you on this.

Reply:
We read these as via `.get(MAP_KEY_INDEX)` etc., but we write them with plain literals that need to match those constants. Here, we should be using
```java
        literals.set(MAP_KEY_INDEX, value.getValue().getStringArray(MAP_KEY_INDEX));
```
Or better yet, using no constants and just use an object like we do in most java code as discussed earlier.

All relatively moot though as long as this function is slated for deletion in the near term.

Reply:
ok got it.